packages/commerce-sdk-react/src/auth/index.test.ts

```diff
@@ -122,6 +122,61 @@ describe('Auth', () => {
         // @ts-expect-error private method
         expect(() => auth.isTokenExpired()).toThrow()
     })
+    test('hasSFRAAuthStateChanged', () => {
+        // Should return true if refresh_token keys are different
+        const auth = new Auth(config)
+        // @ts-expect-error private method
+        auth.set('refresh_token_guest', '123')
+        // @ts-expect-error private method
+        auth.set('refresh_token_registered_copy', '456')
+        // @ts-expect-error private method
+        expect(auth.hasSFRAAuthStateChanged()).toBe(true)
+
+        // Should return true if refresh_token keys are same but values are different
+        // @ts-expect-error private method
+        auth.set('refresh_token_guest', '123')
+        // @ts-expect-error private method
+        auth.set('refresh_token_guest_copy', '456')
+        // @ts-expect-error private method
+        expect(auth.hasSFRAAuthStateChanged()).toBe(true)
+
+        // Should return false if refresh_token keys are same
+        // @ts-expect-error private method
+        auth.set('refresh_token_guest', '123')
+        // @ts-expect-error private method
+        auth.set('refresh_token_guest_copy', '123')
+        // @ts-expect-error private method
+        expect(auth.hasSFRAAuthStateChanged()).toBe(false)
+    })
+    test('isTokenValid', () => {
+        const auth = new Auth(config)
+
+        // Return false if JWT Expired
+        const JWTExpired = jwt.sign({exp: Math.floor(Date.now() / 1000) - 1000}, 'secret')
+        // @ts-expect-error private method
+        auth.set('refresh_token_guest', '123')
+        // @ts-expect-error private method
+        auth.set('refresh_token_guest_copy', '123')
+        // @ts-expect-error private method
+        expect(auth.isTokenValid(JWTExpired)).toBe(false)
+
+        // Return false if SFRA Auth state changed
+        const JWTNotExpired = jwt.sign({exp: Math.floor(Date.now() / 1000) + 1000}, 'secret')
+        // @ts-expect-error private method
+        auth.set('refresh_token_guest', '123')
+        // @ts-expect-error private method
+        auth.set('refresh_token_registered_copy', '456')
+        // @ts-expect-error private method
+        expect(auth.isTokenValid(JWTNotExpired)).toBe(false)
+
+        // Return true if JWT NOT expired and SFRA Auth state NOT changed
+        // @ts-expect-error private method
+        auth.set('refresh_token_guest', '123')
+        // @ts-expect-error private method
+        auth.set('refresh_token_guest_copy', '123')
+        // @ts-expect-error private method
+        expect(auth.isTokenValid(JWTNotExpired)).toBe(true)
+    })
```

Comment:
This block feels like multiple tests grouped together. Is there a logical point at which they could be divided?

Reply:
It's just different branching combinations to test the same condition for different truthy or falsy values based on the scenarios. We don't have enough diverging logic to split them into their own tests.